### ISSUE TITLE
C#: ambiguous overload should produce warning CS1957

### ISSUE THREAD

**Issue by BillWagner** [dotnet/roslyn#25171]
**Version Used**:  

Microsoft Visual Studio Enterprise 2017 
Version 15.5.7
VisualStudio.15.Release/15.5.7+27130.2036
Visual C# 2017   00369-60000-00001-AA058
Microsoft Visual C# 2017

**Steps to Reproduce**:

Use the following code:

```csharp
class Base<T, S>
{
    public virtual string Test(out T x) // CS1957  
    {
        x = default(T);
        return "Base.Test: out";
    }
    public virtual void Test(ref S x) =>
        System.Console.WriteLine("Base.Test: ref");
}

class Derived : Base<int, int>
{
    public override string Test(out int x)
    {
        x = 0;
        return "Derived.Test";
    }

    static void Main()
    {
        int x;
        System.Console.WriteLine(new Derived().Test(out x));
        new Derived().Test(ref x);
    }
}
```

**Expected Behavior**:

The declaration of Derived.Test() should produce a warning, because it could override either overload declared in Base.

**Actual Behavior**:

No warning is produced. The derived version is called. in the first call in Main(), the base version that has a void return in the second. It appears that either the difference in return type, or the difference between 'ref' and 'out' is participating in overload resolution
